src/io/output_mesh.cc

```diff
@@ -60,14 +61,21 @@ OutputMeshBase::OutputMeshBase(Mesh &mesh, const Input::Record &in_rec)
     input_record_(in_rec), 
     orig_mesh_(&mesh),
     max_level_(input_record_.val<int>("max_level")),
-	mesh_type_(MeshType::orig),
     refine_by_error_(input_record_.val<bool>("refine_by_error")),
-    refinement_error_tolerance_(input_record_.val<double>("refinement_error_tolerance"))
+    refinement_error_tolerance_(input_record_.val<double>("refinement_error_tolerance")),
+	el_ds_(nullptr),
+	node_ds_(nullptr)
 {
 }
 
 OutputMeshBase::~OutputMeshBase()
 {
+    if ( (el_ds_!=nullptr) && (mesh_type_ == MeshType::refined)) {
```

Comment:
Comment in which cases we have own distribution and when we just use arrays from computational mesh.